{
	"name": "PL_Throw_Exception",
	"properties": {
		"description": "Provide a simple way of throwing an exception within Data Factory using TSQL error handling.",
		"activities": [
			{
				"name": "Raise Error",
				"description": "Using a SQL database to raise an error/exception but wrapped up as a data factory pipeline. Error message information exposed as a pipeline parameter.",
				"type": "Lookup",
				"dependsOn": [],
				"policy": {
					"timeout": "0.00:10:00",
					"retry": 0,
					"retryIntervalInSeconds": 30,
					"secureOutput": false,
					"secureInput": false
				},
				"userProperties": [],
				"typeProperties": {
					"source": {
						"type": "AzureSqlSource",
						"sqlReaderQuery": {
							"value": "RAISERROR('@{pipeline().parameters.Message}',16,1);",
							"type": "Expression"
						},
						"queryTimeout": "02:00:00",
						"partitionOption": "None"
					},
					"dataset": {
						"referenceName": "DS_ControlFramework",
						"type": "DatasetReference",
						"parameters": {
							"secretName": {
								"value": "@pipeline().globalParameters.controlFrameworkSecretName",
								"type": "Expression"
							},
							"secretURL": {
								"value": "@pipeline().globalParameters.controlFrameworkSecretURL",
								"type": "Expression"
							}
						}
					},
					"firstRowOnly": false
				}
			}
		],
		"parameters": {
			"Message": {
				"type": "string"
			}
		},
		"folder": {
			"name": "Control Framework/Utilities"
		},
		"annotations": [],
		"lastPublishTime": "2022-03-26T14:41:12Z"
	},
	"type": "Microsoft.DataFactory/factories/pipelines"
}